LayoutTests/imported/w3c/web-platform-tests/resources/testdriver-vendor.js

```diff
@@ -151,6 +151,126 @@ async function dispatchTouchActions(actions, options = { insertPauseAfterPointer
     })();`, resolve));
 }
 
+function computeTickDuration(actions, sourceType)
+{
+    sourceToActionWithDurationMap = { "pointer": "pointerMove", "wheel": "scroll" };
+    maxDuration = 0;
+    for (let action of actions) {
+        let duration;
+        if (action.type === "pause" || action.type === sourceToActionWithDurationMap[sourceType])
+            duration = action.duration;
+        if (duration !== undefined) {
+            if (duration > maxDuration)
+                maxDuration = duration;
+        }
+    }
+    return maxDuration;
+}
+
+async function renderingUpdate()
+{
+    await new Promise(requestAnimationFrame);
+    await new Promise(resolve => setTimeout(resolve, 0));
+}
+
+async function waitForScrollCompletion()
+{
+    if (testRunner.isIOSFamily) {
+        await new Promise(resolve => {
+            testRunner.runUIScript(`
+                (function() {
+                    uiController.didEndScrollingCallback = function() {
+                        uiController.uiScriptComplete();
+                    }
+                })()`, resolve);
+        });
+        return renderingUpdate();
+    }
+
+    return new Promise(resolve => {
+        eventSender.callAfterScrollingCompletes(() => {
+            requestAnimationFrame(resolve);
+        });
+    });
+}
+
+async function ensurePresentationUpdate()
+{
+    if (!testRunner.isWebKit2)
+        return renderingUpdate();
+
+    return new Promise(resolve => {
+        testRunner.runUIScript(`
+            uiController.doAfterPresentationUpdate(function() {
+                uiController.uiScriptComplete();
+            });`, resolve);
+    });
+}
+
+/**
+ *
+ * @param {object[]} actions
+ * @returns Promise<undefined>
+ */
+async function dispatchWheelActions(actions)
+{
+    if (!window.eventSender)
+        throw new Error("window.eventSender is undefined.");
+
+    for (let action of actions) {
+        switch (action.type) {
+        case "pause":
+            logDebug(`pause(${action.duration})`);
+            await pause(action.duration);
+            break;
+        case "scroll":
+            // FIXME(261810): Follow tick-based event dispatch logic rather than sending scroll events instantaneously
+            // https://w3c.github.io/webdriver/#dfn-perform-a-scroll
+            let { x, y, origin, deltaX, deltaY, duration } = action;
```

Comment:
What coordinate system are `x` and `y` in?

Reply:
Field | Description
-- | --
x | An unsigned integer representing the pointer x location in viewport coordinates.
y | An unsigned integer representing the pointer y location in viewport coordinates.